src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+  let updatedState = { ...state };
+
+  for (const action of actions) {
+    updatedState = { ...updatedState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(updatedState, action.extraData);
+        statesArray.push(updatedState);
```

Comment:
The `updatedState` object is being pushed to `statesArray` after being modified. However, since `updatedState` is a reference to the same object, all entries in `statesArray` will end up being the same object. Consider cloning `updatedState` before pushing it to `statesArray` to ensure each entry is independent.